### PR TITLE
Warrior update

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

**Comment by kacf** [comment]
@oleorhagen: Most of the stuff here is cherry-picks and already reviewed, but I had to make https://github.com/mendersoftware/meta-mender/pull/957/commits/0aa7dfb876820f82a68fa78f36f5c13a673a6f55 to keep warrior working with the old client name. Does it look ok to you?

File: meta-mender-core/recipes-mender/mender/mender.inc [hunk 1]
```diff
@@ -244,6 +244,16 @@ do_install() {
...
 
+do_install_append_mender-systemd() {
+    # Versions of Mender client after 2.2 come with the service file called
+    # "mender-client" by default, but we don't want to change this in a stable
+    # Yocto branch, so for warrior and older we do this rename back to the old
+    # name "mender".
+    if [ -e ${D}${systemd_unitdir}/system/mender-client.service ]; then
+        mv ${D}${systemd_unitdir}/system/mender-client.service ${D}${systemd_unitdir}/system/mender.service
+    fi
+}
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender/mender.inc:255]: Note to self: Add changelog for this.

**Comment by oleorhagen** [comment]
My lord this renaming turned out to be a pain

**Comment by eigendude** [comment]
I'm trying to boot mender on a BeagleBone Black, and the commit `mender-grub: Dynamically determine mender_grub_storage_device.` seems to break the boot process. On the latest warrior branch of meta-mender, I get the error:

```
error: no such device: ((hd0,gpt1)/efi/boot)/EFI/BOOT/grub.cfg.
lock: OK
lock: OK
error: can't find command `regexp'.

error: disk `,msdos2' not found.

Dropping to grub prompt for unknown reason. Should never get here.
```

When I revert the commit, the output is:

```
Welcome to GRUB!

error: no such device: ((hd0,gpt1)/efi/boot)/EFI/BOOT/grub.cfg.
lock: OK
lock: OK
error: can't find command `regexp'.

[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.3.13-jumpnow (oe-user@oe-host) (gcc version 8.3.0 (GCC)) #1 Fri Jun 12 19:
29:54 UTC 2020
[    0.000000] CPU: ARMv7 Processor [413fc082] revision 2 (ARMv7), cr=10c5387d
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: TI AM335x BeagleBone Black
...
```

And the device boots into linux successfully (though I've misconfigured the RTC and linux gets stuck 2.2 seconds into boot).

When I do a hard-reset to the PR directly before this (https://github.com/mendersoftware/meta-mender/pull/941), the regexp error goes away entirely:

```
Welcome to GRUB!

error: no such device: ((hd0,gpt1)/efi/boot)/EFI/BOOT/grub.cfg.
lock: OK
lock: OK

[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.3.13-jumpnow (oe-user@oe-host) (gcc version 8.3.0 (GCC)) #1 Fri Jun 12 19:
29:54 UTC 2020
[    0.000000] CPU: ARMv7 Processor [413fc082] revision 2 (ARMv7), cr=10c5387d
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: TI AM335x BeagleBone Black
...
```

My guess is that version 2.02 of the the pre-compiled `grub-efi-bootarm.efi` was built without the regexp module. I haven't tried the latest 2.04.

**Comment by eigendude** [comment]
Confirmed that 2.04 on the Zeus branch works:

```
Welcome to GRUB!

lock: OK
lock: OK


EFI stub: Booting Linux Kernel...
EFI stub: Using DTB from configuration table
EFI stub: Exiting boot services and installing virtual address map...
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.6.18-jumpnow (oe-user@oe-host) (gcc version 9.2.0 (GCC)) #1 Sun Jun 14 17:11:07 UTC 2020
[    0.000000] CPU: ARMv7 Processor [413fc082] revision 2 (ARMv7), cr=10c5387d
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: TI AM335x BeagleBone Black

...
```

**Comment by kacf** [comment]
> My guess is that version 2.02 of the the pre-compiled `grub-efi-bootarm.efi` was built without the regexp module. I haven't tried the latest 2.04 version of `grub-efi-bootarm.efi`.

I believe you're right. I can take a look at this tomorrow.

**Comment by kacf** [comment]
@eigendude: Check out #1010!

**Pull request merged**